packages/components/src/components/checkbox-wrapper/checkbox-wrapper.tsx

```diff
@@ -47,11 +50,28 @@ export class CheckboxWrapper {
   /** Show or hide label. For better accessibility it's recommended to show the label. */
   @Prop() public hideLabel?: BreakpointCustomizable<boolean> = false;
 
+  /** Disables the checkbox and shows a loading indicator. */
+  @Prop() public loading?: boolean = false;
+
   /** Adapts the color depending on the theme. */
   @Prop() public theme?: Theme = 'light';
 
   private input: HTMLInputElement;
 
+  @Listen('click', { capture: true })
+  public onClick(e: MouseEvent): void {
+    if (isDisabledOrLoading(this.input.disabled, this.loading)) {
+      e.preventDefault();
+    }
+  }
+
+  @Listen('keydown')
+  public handleKeyDown(e: KeyboardEvent): void {
```

Comment:
to be in synch with other handler
```suggestion
  public onKeydown(e: KeyboardEvent): void {
```